### PR TITLE
Fix panic on deserializing partial guild with no Nitro boosters.

### LINKED ISSUE #623: http::get_guild panics on guilds without Nitro boosters
I have a guild without any Nitro boosters, and trying to http::get_guild it panics the JSON decoder. This is because the JSON object contains `"premium_subscription_count": null`, while the field in the Rust struct is defined as `pub premium_subscription_count: u64`.

The error itself is `Json(Error("invalid type: null, expected u64", line: 1, column: 239))`, and sure enough if I `curl` the endpoint myself, column 239 corresponds to `"premium_subscription_count": null`.

I'm working on a PR for this myself, but I thought I'd log the issue beforehand in case I forget/am not able to fix it.

EDIT: This is on Serenity version `0.6.0-rc.2.1`, rustc version `rustc 1.35.0 (3c235d560 2019-05-20)`

### PR DESCRIPTION
**Pull request by xSke**
Closes #623 .

### CODE REVIEW

**Comment by xSke** [comment]
The solution here means the deserialization/serialization *technically* isn't round-trip (input null would get reserialized to 0), I'm not sure whether this is important to you or not. I didn't want to directly make the field an `Option<u64>`, since that's a breaking API change for no practical benefit (`None` and `Some(0)` would be equivalent in practice).

File: src/model/guild/partial_guild.rs [hunk 1]
```diff
@@ -39,6 +39,8 @@ pub struct PartialGuild {
     pub verification_level: VerificationLevel,
     pub description: Option<String>,
     pub premium_tier: PremiumTier,
+    // In some cases Discord returns `null` rather than 0
```
Comment (Lakelezz) [inline src/model/guild/partial_guild.rs]: Missing fullstop at the end of the sentence.

File: src/model/utils.rs [hunk 2]
```diff
@@ -231,6 +231,11 @@ pub fn deserialize_u64<'de, D: Deserializer<'de>>(deserializer: D) -> StdResult<
     deserializer.deserialize_any(U64Visitor)
 }
 
+/// Deserializes an u64, returning 0 if the deserialization failed
```
Comment (Lakelezz) [inline src/model/utils.rs]: Missing fullstop at the end of the sentence.

**Comment by Lakelezz** [comment]
Thank you :)

**Pull request merged**